src/Tools.pm

```diff
@@ -484,42 +524,26 @@ returns 1 if yes, 0 if no
 
 =cut
 
-sub IsDMRaidSlave {
-    my $disk = shift;
-
-    unless ( DMRaidAvailable() and -e $dmsetup) {
-        return 0;
-    }
+sub IsDMRaidSlave
+{
+  my $self = $lib_ref;
+  my $dev = shift;
 
-    my $majmin_disk = `stat -c "%t:%T" $disk`;
-    chomp($majmin_disk);
-    my @dmparts = ();
+  my $r = $self->{cache}{IsDMRaidSlave}{$dev};
```

Comment:
As this is really a port and not new code, I'd rather leave it as is. But of course, you are right.